grpc-native/src/main/java/org/ballerinalang/net/grpc/protobuf/cmd/GrpcCmd.java

```diff
@@ -130,14 +128,40 @@ public void execute() {
             return;
         }
 
-        // check input protobuf file path
-        Optional<String> pathExtension = getFileExtension(protoPath);
-        if (pathExtension.isEmpty() ||
-                !PROTO_EXTENSION.equalsIgnoreCase(pathExtension.get())) {
-            String errorMessage = "Invalid proto file path. Please input valid proto file location.";
-            outStream.println(errorMessage);
-            return;
+        File input = new File(protoPath);
+        if (input.isDirectory()) {
+            // Multiple proto files
+            File[] protoFiles = input.listFiles();
+            if (protoFiles == null) {
+                String errorMessage = "Input proto files directory is empty. Please input valid proto file location.";
+                outStream.println(errorMessage);
+            } else {
+                for (File protoFile : protoFiles) {
+                    // check input protobuf file path
+                    Optional<String> pathExtension = getFileExtension(protoFile.getPath());
+                    if (!PROTO_EXTENSION.equalsIgnoreCase(pathExtension.get())) {
+                        String errorMessage = "Invalid proto file path: " + protoFile.getPath() +
+                                ". Please input valid proto file location.";
+                        outStream.println(errorMessage);
+                        return;
```

Comment:
here we are restricting to have only proto files in the directory. instead, shall we filter only proto files and proceed.

Reply:
Shall we also test the scenario where we have a dependent proto file also in the same directory. want to make sure we do not regenerate it again.

Reply:
> here we are restricting to have only proto files in the directory. instead, shall we filter only proto files and proceed.

Updated in 832d112


> Shall we also test the scenario where we have a dependent proto file also in the same directory. want to make sure we do not regenerate it again.

Added a test case with dependent proto in a4372b1